android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleTagManagerBridge.java

```diff
@@ -131,6 +179,70 @@ private DataLayer getDataLayer() {
         return mDatalayer;
     }
 
+    private String generateFunctionCallTagEventName(String functionName) {
+        return FUNCTON_TAG_EVENT_PREFIX + functionName;
```

Comment:
I'm kind of unclear on why we need to prefix function names. You said something about avoiding collisions. Is this collisions with other **native** libraries which also use GTM, or collisions on events sent from JS<->native? I am not really sure this is a problem?

Reply:
This is merely an attemp to namespace the functions. Take a look at [this module](https://github.com/reyhoonia/react-native-google-analytics-bridge/blob/079c1f5359d30bea7b4d8ec99dd2c0bfcc35921e/src/Helpers/FunctionTagHandler/index.android.js#L19).

```javascript
const eventName = GoogleTagManagerBridge.TAG_EVENT_PREFIX + functionName;
return GoogleTagManagerBridge.registerFunctionCallTagHandler(functionName)
  .then(() => {
    DeviceEventEmitter.addListener(eventName, function (event) {
      handler(this._fn, event);
    }.bind({_fn: functionName}));
    return true;
  });
```
If we don't prefix the function name then it becomes more likely to register a function name that may match a native event emitted by DeviceEventEmitter. Unless the developers have full knowledge of events that can be dispatched via DeviceEventEmitter, wether its native to RN or introduced by other 3rd-party libraries.